packages/nimble-components/src/wafer-map/tests/DataGenerator.ts

```diff
@@ -0,0 +1,56 @@
+import type { WaferMapDie } from '../types';
+import type { IValueGenerator } from './ValueGenerator';
+
+const valueToString = (value: number): string => {
+    return value % 1 ? value.toFixed(2) : Math.trunc(value).toString();
+};
+
+export const generateWaferData = (
+    numDies: number,
+    valueGenerator?: IValueGenerator
```

Comment:
instead of making this nullable you could have a `defaultValueGenerator` instead
```ts
valueGenerator = defaultValueGenerator
```

Reply:
I did consider this, however I would prefer to keep it as it is since I think it makes the method easier to use - I wanted to emphasize that the value generator is optional. Internally, if no value generator is provided, Math.random() is used for each individual die

Reply:
Ah yea, what I was describing was using [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters)

So if the function is defined as:

```ts
export const generateWaferData = (
    numDies: number,
    valueGenerator: IValueGenerator = defaultValueGenerator
): WaferMapDie[] => {
```

then it can be invoked without a parameter (as it already has a default):
```ts
generateWaferData(100);
```

Or with a parameter:
```ts
generateWaferData(100, goodValueGenerator(seed));
```

You still get the behavior of optional parameters but without needing to add unnecessary null checks. It's in a test helper so I won't require the change. Just wanted to share the feature and benefits.